### ISSUE TITLE
Caught Network I/O errors on 127.0.0.1:2424, trying an automatic reconnection...

### ISSUE THREAD

**Issue by mikayeln** [orientechnologies/orientdb#7392]
### OS: Linux

We are running large java ee application and using orientdb 2.2.19.

On client side we are receiving a lot of warnings like below 

09:35:04,245 INFO  [com.orientechnologies.orient.client.remote.OStorageRemote] (TaskInvokerRegular-3[__MAIN_WS__]) Caught Network I/O errors on 127.0.0.1:2424/vis_demo, trying an automatic reconnection... (error: Connection reset)

09:35:02,160 INFO  [com.orientechnologies.orient.client.remote.OStorageRemote] (TaskInvokerRegular-3[__MAIN_WS__]) Caught Network I/O errors on 127.0.0.1:2424/vis_demo, trying an automatic reconnection... (error: Channel is closed)

Reconnection tooks 100-200ms in average. It's dramatically affects the performance of application.

Do you have any suggestions how to fix or improve the situation?

**Comment by lvca** [comment]
Can you see in the server logs why there is this disconnection? Could it also possible that your commands are so long that cause a timeout. In this case, you can adjust the connection timeout.

What kind of operation are you executing?

**Comment by mikayeln** [comment]
unfortunately there are no any messages (Error, info or warn) on server side. 
My commands are just regular fast queries like select or update.

**Comment by mikayeln** [comment]
Here is server configs

`
<properties>
        <entry value="100" name="db.pool.min"/>
        <entry value="5000" name="db.pool.max"/>
	<entry value="100" name="script.pool.maxSize"/>
        <entry value="false" name="storage.keepOpen"/>
        <entry value="false" name="profiler.enabled"/>
        <entry value="200000" name="network.maxConcurrentSessions"/>
        <entry value="60" name="network.http.sessionExpireTimeout"/>
        <entry value="10000" name="server.channel.cleanDelay"/>
        <entry value="5000" name="network.socketTimeout"/>
        <entry value="false" name="server.openAllDatabasesAtStartup"/>
        <entry value="5" name="network.token.expireTimeout"/>
    </properties>
`

**Comment by mikayeln** [comment]
I've set the FINEST log level for server. On the time of message in client log on server side there was following log

```
2017-05-04 16:00:29:055 CONFI Remote client connected from: OClientConnection [id=140, source=/127.0.0.1:48752, since=1493892029055] [OClientConnectionManager]
2017-05-04 16:00:29:056 FINE  Disconnecting connection OClientConnection [id=140, source=/127.0.0.1:48752, since=1493892029055]... [OClientConnectionManager]
2017-05-04 16:00:29:056 FINE  Disconnected connection OClientConnection [id=140, source=/127.0.0.1:48752, since=1493892029055] found 1 channels [OClientConnectionManager]
2017-05-04 16:00:29:056 FINE  Disconnecting connection OClientConnection [id=140, source=/127.0.0.1:48752, since=1493892029055]... [OClientConnectionManager]
2017-05-04 16:00:29:056 FINE  Disconnected connection OClientConnection [id=140, source=/127.0.0.1:48752, since=1493892029055] found 0 channels [OClientConnectionManager]
2017-05-04 16:00:29:056 FINE  Sent run-time exception to the client /127.0.0.1:48752: com.orientechnologies.orient.enterprise.channel.binary.OTokenSecurityException: The token provided is expired [ONetworkProtocolBinary]
```

**Comment by lvca** [comment]
Could you please try the same test but by removing all your custom settings?

**Comment by tglman** [comment]
hi @mikayeln,

Seeing your second log seems the token is expired so need to be renegotiated, this may be due to the fact that the live time of the token is just five minutes: `<entry value="5" name="network.token.expireTimeout"/>` is it there any reason for this timeout? can you increase it? The default one is one hour.

Regards

**Comment by mikayeln** [comment]
This setting was set to 5 to solve (temp solution) another issue with non closing network connections. I will give a try to 60 minutes.

**Comment by careerscale** [comment]
We see this with small application

**Comment by ozgursucu** [comment]
Hi @tglman I am also experiencing this issues and sometimes when connection is lost the client side waits forever.

**Comment by tglman** [comment]
Hi,

Issues related to this were fixed in 2.2.30, closing it.
